File: Backend/PRScapstone/Models/Vendor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace PRScapstone.Models
{
    public class Vendor
    {
        public int Id { get; set; }
        [Index(IsUnique = true)]
        [StringLength(10)]
        [Required]
        public string Code { get; set; }
        [Required][StringLength(255)]
        public string Name { get; set; }
        [Required][StringLength(255)]
        public string Address { get; set; }
        [Required][StringLength(255)]
        public string City { get; set; }
        [Required][StringLength(2)]
        public string State { get; set; }
        [Required][StringLength(5)]
        public string Zipcode { get; set; }
        [Required][StringLength(12)]
        public string Phone { get; set; }
        [Required][StringLength(100)]
        public string Email { get; set; }
        [Required]
        public bool PreApproved { get; set; }
        [Required]
        public bool Active { get; set; }
        //public DateTime DateCreated { get; set; }
        //public DateTime DateUpdated { get; set; }

    }
}

File: Frontend/src/app/models/vendor.ts
export class Vendor {
  Id: number;
  Code: string;
  Name: string;
  Address: string;
  City: string;
  State: string;
  Zipcode: string;
  Phone: string;
  Email: string;
  PreApproved: boolean;
  Active: boolean;

  constructor(
    Id: number,
    Code: string,
    Name: string,
    Address: string,
    City: string,
    State: string,
    Zipcode: string,
    Phone: string,
    Email: string,
    PreApproved: boolean,
    Active: boolean
  ) {
  this.Id = Id;
  this.Code = Code;
  this.Name = Name;
  this.Address = Address;
  this.City = City;
  this.State = State;
  this.Zipcode = Zipcode;
  this.Phone = Phone;
  this.Email = Email;
  this.PreApproved = PreApproved;
  this.Active = Active;
  }
}
